workspace/src/string-transform.js

```diff
@@ -38,13 +38,139 @@ function transform_ins_ins(op0, op1) {
+    return op1;
+  }
+
+  // case 1: destination of Mov located in the deletion area
+  // case 1.1:
+  //     ______________
+  //    |              |
+  //    |              v
+  //   [_____]     [--------]
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //    op1           op0(del)
+
+  // case 1.2:
+  //              _______________
+  //             |               |
+  //             v               |
+  //        [--------]       [_____]
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //         op0(del)         op1
+  if ((op1.pos+op1.length<=op0.pos && op1.des >= op0.pos && op1.des <= op0.pos+op0.length)
+    ||(op1.pos>=op0.pos+op0.length && op1.des>=op0.pos && op1.des<=op0.pos+op0.length)){
+    return makeOpMov(op1.pos,op1.length,op0.pos);
+  }
+
+  // case 2: destination of Mov located after deletion area
+  // case 2.1:
+  //     _______________________
+  //    |                       |
+  //    |                       v
+  //   [_____]     [--------]
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //    op1           op0(del)
+
+  // case 2.2:
+  //                    _______
+  //                   |       |
+  //                   v       |
+  //        [--------]       [_____]
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //         op0(del)         op1
+  if (op1.pos+op1.length<=op0.pos && op1.des >= op0.pos+op0.length){
+    return makeOpMov(op1.pos,op1.length,op1.des-op0.length);
+  }
+
+  // case 3: Partial Overlap between Mov area and Del area
+  // case 3.1:
+  //     ____________
+  //    |            |
+  //    |            V
+  //   {____[===}------]
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //    op1       op0(del)
+  //
+  //           _______
+  //          |       |
+  //          v       |
+  //       [____{===]------}
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //        op0(del)   op1
+
+  if((op1.pos<=op0.pos && op1.pos+op1.length<=op0.pos+op0.length&& op1.pos+op1.length >= op0.pos
+    && op1.des<=op0.pos+op0.length && op1.des>=op1.pos) || (op1.pos>=op0.pos && op1.pos<=op0.pos+op0.length
+    && op1.pos+op1.length>=op0.pos+op0.length && op1.des>=op0.pos && op1.des<=op1.pos)){
+    return makeOpMov(0,0,0); // same as it doesn't move
+  }
+
+  // case 3.2:
+  //  __
+  // |  |
+  // |  |
+  // v {____[===}------]
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //    op1        op0(del)
+
+  if (op1.pos<=op0.pos && op1.pos+op1.length<=op0.pos+op0.length&& op1.pos+op1.length >= op0.pos && op1.des<=op0.pos){
+    return makeOpMov(op1.pos,op0.pos-op1.pos,op1.des);
+  }
+
+  // case 3.3:
+  //     ___________________
+  //    |                   |
+  //    |                   V
+  //   {____[===}------]
+  //  0 1 2 3 4 5 6 7 8 9 10 11 12 13 ...
+  //    op1       op0(del)
+  if (op1.pos<=op0.pos && op1.pos+op1.length<=op0.pos+op0.length&& op1.pos+op1.length >= op0.pos && op1.des>=op0.pos+op0.length){
```

Comment:
all of these logics can be combined. 

1. Use single layer of `if` `else if`
2. combine cases
3. maybe assign some of the conditions to a variable.